subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy

```diff
@@ -694,13 +695,60 @@ abstract class MakeGreen implements TransformAction<TransformParameters.None> {
         fails(":a:resolve")
 
         then:
-        // Documents existing behaviour. Should fail eagerly and with a better error message
-        failure.assertHasDescription("Execution failed for task ':a:resolve'.")
-        failure.assertHasCause("Execution failed for MakeGreen: ${file('b/build/b.jar')}.")
-        failure.assertHasCause("No service of type ${propertyType} available.")
+        if (configurationTime) {
+            failure.assertHasDescription("A problem occurred evaluating root project")
+            failure.assertHasCause("Cannot register artifact transform MakeGreen with parameters null")
```

Comment:
Why is this one emitted?

Reply:
The error happens when registering the transform, so we add this cause here: https://github.com/gradle/gradle/blob/a46501d7e99ec8f4878916ebe36186a6607ac7c1/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultVariantTransformRegistry.java#L77

Reply:
I find this  `Cannot register artifact transform MakeGreen with parameters null` message problematic. `with parameters null` is misleading in that this is a parameter-less transform. We should tell the user that instead.

But maybe we shouldn't even talk about parameters in the first place. Why don't we use `from` and `to` attributes to help the user identify the problematic transform instead?

Reply:
For identifying the transform, isn't the project and the line in the build script enough?
We probably should not show `with parameters null` but `without parameters` as you suggest.

Reply:
I would really like us to reconsider printing anything about the parameters. If it is the build script line that is the primary identifier, then we still don't need the parameters. That said, transforms are not necessarily registered from the build script. Indeed, they are more likely to be registered from a plugin like AGP. In any case, I think `to` and `from` might be useful to be printed; parameters not.

Reply:
I wonder how interesting the validation messages are if the plugin registers the transform. If the class is invalid since it doesn't use the right types for annotation, then the plugin will never work.
I'll remove the parameters since they won't have a good `toString` method anyway and see what I can do about the from and to attributes.

Reply:
Now we show from/to attributes and no parameters.